packages/flutter_tools/test/general.shard/build_system/targets/localizations_test.dart

```diff
@@ -72,6 +74,52 @@ void main() {
     expect(processManager.hasRemainingExpectations, false);
   });
 
+  testUsingContext('generateLocalizations throws exception on missing flutter: generate: true flag', () async {
+    final FileSystem fileSystem = MemoryFileSystem.test();
+    final Logger logger = BufferLogger.test();
+    final FakeProcessManager processManager = FakeProcessManager.list(<FakeCommand>[]);
+    final Directory arbDirectory = fileSystem.directory('arb')
+      ..createSync();
+    arbDirectory.childFile('foo.arb').createSync();
+    arbDirectory.childFile('bar.arb').createSync();
+
+    // Missing flutter: generate: true should throw exception.
+    fileSystem.file('pubspec.yaml').writeAsStringSync('''
+flutter:
+  uses-material-design: true
+''');
+
+    final LocalizationOptions options = LocalizationOptions(
+      header: 'HEADER',
+      headerFile: Uri.file('header'),
+      arbDirectory: Uri.file('arb'),
+      deferredLoading: true,
+      outputClass: 'Foo',
+      outputLocalizationsFile: Uri.file('bar'),
+      preferredSupportedLocales: 'en_US',
+      templateArbFile: Uri.file('example.arb'),
+      untranslatedMessagesFile: Uri.file('untranslated'),
+      // Set synthetic package to true.
+      useSyntheticPackage: true,
+    );
+
+    try {
+      await generateLocalizations(
```

Comment:
`expect(() => ..., throwsA(isA<Exception>())`

Reply:
you should also check that the logger error text contains some part of the message you added